src/common.jl

```diff
@@ -105,3 +105,16 @@ function regularize_symmat!(A::Matrix{T}, lambda::Real) where T<:Real
     return A
 end
 
+"""
+    calcscattermat(Z::DenseMatrix, covarianceestimator = nothing)
+
+Calculate the scatter matrix of centered data `Z`. If `covarianceestimator` is
+not `nothing` then it is used to calculate the covariance matrix.
+"""
+function calcscattermat(Z::DenseMatrix, covarianceestimator = nothing)
+    if covarianceestimator === nothing
```

Comment:
It would be more idiomatic to use dispatch (with a positional argument). EDIT: if you use `SimpleCovarianceEstimator` I guess you don't need that function at all.

Reply:
Agree, instead of passing `nothing`, use a default estimator.